packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -338,35 +307,35 @@ class VisualStudio {
   ///
   /// Returns false if the required information is missing since older versions
   /// of Visual Studio might not include them.
-  bool installationHasIssues(Map<String, dynamic>installationDetails) {
+  bool installationHasIssues(VswhereDetails installationDetails) {
     assert(installationDetails != null);
-    if (installationDetails[_isCompleteKey] != null && !(installationDetails[_isCompleteKey] as bool)) {
+    if (!(installationDetails.isComplete ?? true)) {
       return true;
     }
 
-    if (installationDetails[_isLaunchableKey] != null && !(installationDetails[_isLaunchableKey] as bool)) {
+    if (!(installationDetails.isLaunchable ?? true)) {
       return true;
     }
 
-    if (installationDetails[_isRebootRequiredKey] != null && installationDetails[_isRebootRequiredKey] as bool) {
+    if (installationDetails.isRebootRequired ?? false) {
       return true;
     }
 
     return false;
   }
 
-  /// Returns the details dictionary for the latest version of Visual Studio
-  /// that has all required components and is a supported version, or {} if
-  /// there is no such installation.
+  /// Returns the details for the latest version of Visual Studio that has all
+  /// the required components and is a supported version, or null if there is no
+  /// such installation.
   ///
-  /// If no installation is found, the cached VS details are set to an empty map
-  /// to avoid repeating vswhere queries that have already not found an installation.
```

Comment:
This comment was previously incorrect. The "cached VS details" was only set if an installation was found but had issues.